backend/src/apiserver/storage/pipeline_store_kubernetes.go

```diff
@@ -0,0 +1,502 @@
+		)
+	} else if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to create the pipeline")
+	}
+
+	err = k.client.Status().Update(context.TODO(), &k8sPipeline)
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to set the pipeline status")
+	}
+
+	return k8sPipeline.ToModel(), nil
+}
+
+func (k *PipelineStoreKubernetes) UpdatePipelineStatus(pipelineId string, status model.PipelineStatus) error {
+	// Do nothing. Just return nil to avoid show an unrelated error
+	return nil
+}
+
+func (k *PipelineStoreKubernetes) UpdatePipelineVersionStatus(pipelineVersionId string, status model.PipelineVersionStatus) error {
+	k8sPipelineVersion, err := k.getK8sPipelineVersion(pipelineVersionId)
+	if err != nil {
+		return err
+	}
+
+	conditionSet := false
+
+	for i := range k8sPipelineVersion.Status.Conditions {
+		condition := &k8sPipelineVersion.Status.Conditions[i]
+
+		if condition.Type == "PipelineVersionStatus" {
+			if condition.Reason == string(status) && condition.Message == condition.Reason {
+				return nil
+			}
+
+			condition.Reason = string(status)
+			condition.Message = string(status)
+			condition.Status = metav1.ConditionTrue
+
+			conditionSet = true
+
+			break
+		}
+	}
+
+	if !conditionSet {
+		k8sPipelineVersion.Status.Conditions = append(k8sPipelineVersion.Status.Conditions, v2beta1.SimplifiedCondition{
+			Type:    "PipelineVersionStatus",
+			Reason:  string(status),
+			Message: string(status),
+			Status:  metav1.ConditionTrue,
+		})
+	}
+
+	// TODO: Add resilience to resource version changing
+	err = k.client.Status().Update(context.TODO(), k8sPipelineVersion)
+	if err != nil {
+		return util.NewInternalServerError(err, "Failed to update the pipeline version status")
+	}
+
+	return nil
+}
+
+func (k *PipelineStoreKubernetes) CreatePipelineVersion(pipelineVersion *model.PipelineVersion) (*model.PipelineVersion, error) {
+	pipeline, err := k.GetPipeline(pipelineVersion.PipelineId)
+	if err != nil {
+		return nil, err
+	}
+
+	return k.createPipelineVersionWithPipeline(pipeline, pipelineVersion)
+}
+
+func (k *PipelineStoreKubernetes) UpdatePipelineDefaultVersion(pipelineId string, versionId string) error {
+	return util.NewBadRequestError(errors.New("pipeline default version is unsupported"),
+		"pipeline default version is unsupported when storing in Kubernetes")
+}
+
+func (k *PipelineStoreKubernetes) GetLatestPipelineVersion(pipelineId string) (*model.PipelineVersion, error) {
+	k8sPipelineVersions, err := k.getK8sPipelineVersions()
+	if err != nil {
+		return nil, err
+	}
+
+	var latestK8sPipelineVersion *v2beta1.PipelineVersion
+
+	for _, k8sPipelineVersion := range k8sPipelineVersions.Items {
+		if !k8sPipelineVersion.IsOwnedByPipeline(pipelineId) {
+			continue
+		}
+
+		if latestK8sPipelineVersion == nil || k8sPipelineVersion.CreationTimestamp.Time.After(latestK8sPipelineVersion.CreationTimestamp.Time) {
+			latestK8sPipelineVersion = &k8sPipelineVersion
+
+			continue
+		}
+	}
```

Comment:
optionally you could maybe simplify this by using sort pkg, maybe something like this: 

```golang
	sort.SliceStable(k8sPipelineVersions.Items, func(i, j int) bool {
		return k8sPipelineVersions.Items[i].CreationTimestamp.After(k8sPipelineVersions.Items[j].CreationTimestamp.Time)
	})
	latestK8sPipelineVersion = &k8sPipelineVersions.Items[0]
```

Reply:
This leads to return any pipeline version no matter if is owned by the pipeline ID provided in the function.